javawrapper/device/samples/direct call of wrapped functions/Main.java

```diff
@@ -0,0 +1,164 @@
+package javaWrapper;
+
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.sun.jna.Pointer;
+import com.sun.jna.Structure;
+import com.sun.jna.ptr.IntByReference;
+
+import javaWrapper.Iothub_client_wrapperLibrary.AMQP_Protocol;
+import javaWrapper.Iothub_client_wrapperLibrary.HTTP_Protocol;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_CLIENT_HANDLE;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_CLIENT_RESULT;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_MESSAGE_HANDLE;
+import javaWrapper.Iothub_client_wrapperLibrary.IotHubEventCallback;
+import javaWrapper.Iothub_client_wrapperLibrary.IotHubMessageCallback;
+import javaWrapper.Iothub_client_wrapperLibrary.MAP_HANDLE;
+
+
+
+
+
+public class Main {
+
+	public static int callbackCounter = 0;
+	
+	public static IOTHUB_CLIENT_HANDLE handle = new IOTHUB_CLIENT_HANDLE(null);
+	public static IOTHUB_MESSAGE_HANDLE messageHandle;
+	public static int IOTHUB_CLIENT_OK = 0;
+	static final int MESSAGE_COUNT = 5;
+	public static EVENT_INSTANCE[] messages = new EVENT_INSTANCE[MESSAGE_COUNT];
+	static Pointer receiveContext;
+	
+	
+	public static Iothub_client_wrapperLibrary IoTHub = Iothub_client_wrapperLibrary.INSTANCE;
+	
+	
+	public static void main(String[] args) throws InterruptedException 
+	{
+		String connectionString = "HostName=javaTest.azure-devices.net;DeviceId=Device1;SharedAccessKey=pXuifS+EUVY7oewL1hGpttpq5r/jmgONjn41xPK5knM=";
+		
+		System.out.println("library loaded aaaa");
+		
+		HTTP_Protocol protocol = new HTTP_Protocol();
+		//AMQP_Protocol protocol = new AMQP_Protocol();
+			
+		handle = IoTHub.IoTHubClient_CreateFromConnectionString(connectionString, protocol);
+		
+		IntByReference timeout = new IntByReference(241000);
+		if (IoTHub.IoTHubClient_SetOption(handle.getPointer(), "timeout", timeout) != IOTHUB_CLIENT_OK)
+	    {
+			System.out.println("failure to set option \"timeout\"\r\n");
+	    }
+	
+		IntByReference minimumPollingTime = new IntByReference(9);
+	    if (IoTHub.IoTHubClient_SetOption(handle.getPointer(), "MinimumPollingTime", minimumPollingTime.getPointer()) != IOTHUB_CLIENT_OK)
+	    {
+	    	System.out.println("failure to set option \"MinimumPollingTime\"\r\n");
+	    }
+	    
+	    if (IoTHub.IoTHubClient_SetMessageCallback(handle.getPointer(), ReceiveMessageCallback, receiveContext) != IOTHUB_CLIENT_OK)
+        {
+            System.out.println("ERROR: IoTHubClient_LL_SetMessageCallback..........FAILED!\r\n");
+        }
+        else
+        {
+		    for (int i = 0; i < MESSAGE_COUNT; i++)
+	        {
+		    	messages[i] = new EVENT_INSTANCE();
+			    String message = "test message";
+
+			    if ((messages[i].messageHandle = IoTHub.IoTHubMessage_CreateFromString(message)) == null)
+			    {
+			        System.out.println("ERROR: iotHubMessageHandle is NULL!\r\n");
+			    }
+			    else
+			    {
+			    	MAP_HANDLE propMap;
+			    	
+			    	messages[i].messageTrackingId = i;
+			    	messages[i].write();
```

Comment:
I need it to write the fields of the structure to native memory.
You will have to manually call Structure.write/read before/after your native function call, since JNA will not know that the pointer reference a structure that need to be synched with native memory